### DISCUSSION TITLE
Fluent User Defined Functions using Python

### DISCUSSION THREAD

**Discussion by abhishekchitwar** [ansys/pyfluent#292] [Ideas]
Hi Team!
### Idea is to support User-defined functions (UDFs) written in Python under PyAnsys-Pyfluent ecosystem.

**Comment by mkvani** [comment]
This seem to be a good idea. This will allow use of single programing language. Many Python programmers will find it easier.

**Comment by dnwillia-work** [comment]
@abhishekchitwar Interesting thought, but probably a natural thing to ask about.    Would you like the have the Python function executing in Fluent like a "C" UDF does today?  i.e. in Fluent's host process and compute nodes?

If so, this could be quite a big job.  It requires embedding a compatible Python interpreter directly inside the Fluent host and compute nodes, and defining APIs running locally inside Fluent to do things like you can do in UDFs currently (eg: loop over threads, cells and faces).  

So, it would be very helpful if you could be as specific as possible about what you would like to do in such a UDF.   Extending Fluent's expression language might be the better choice here depending on what it is you want to do.
